### PR TITLE
Release v1.5.0

### PR DESCRIPTION
**Pull request by DylanYang0523**
### PR Reminders
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

File: contracts/interface/IOtcMakerEvent.sol [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+interface IOtcMakerEvent {
+    event UpdateCaller(address oldCaller, address newCaller);
```
Comment (tailingchen) [inline contracts/interface/IOtcMakerEvent.sol:6]: follow name convention -> CallerUpdated

File: contracts/otcMaker/OtcMaker.sol [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+    function setCaller(address newCaller) external override onlyOwner {
+        // OM_ZA: zero address
+        require(newCaller != address(0), "OM_ZA");
+        _caller = newCaller;
+        emit UpdateCaller(_caller, newCaller);
+    }
+
+    function setMarginRatioLimit(uint24 marginRatioLimitArg) external override onlyOwner {
+        require(marginRatioLimitArg > 62500 && marginRatioLimitArg < 1000000, "OM_IMR");
```
Comment (tailingchen) [inline contracts/otcMaker/OtcMaker.sol:62]: Is marginRatioLimitArg limited within 10%?
If yes, should modify `1000000` to `100000`.

File: contracts/otcMaker/OtcMaker.sol [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        _marginRatioLimit = marginRatioLimitArg;
+    }
+
+    function deposit(address token, uint256 amount) external override onlyOwner {
```
Comment (tailingchen) [inline contracts/otcMaker/OtcMaker.sol:66]: To make the responsibilities more clear, suggest to introduce another role, `fundOwner`, to manage fund.
- Owner: only responsible for contract upgrade
- FundOwner: manage fund and positions in Curie.
- Caller: openPositionFor

File: contracts/interface/IOtcMakerStruct.sol [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (tailingchen) [inline contracts/interface/IOtcMakerStruct.sol:1]: Change the license to `GPL-3.0-or-later`

File: contracts/interface/IOtcMaker.sol [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    function claimWeek(
+        address merkleRedeem,
+        address liquidityProvider,
+        uint256 week,
+        uint256 claimedBalance,
+        bytes32[] calldata _merkleProof
+    ) external;
+
+    function setCaller(address newCaller) external;
```
Comment (tailingchen) [inline contracts/interface/IOtcMaker.sol:40]: remove admin function from interface.

File: contracts/interface/IOtcMaker.sol [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    function setMarginRatioLimit(uint24 openMarginRatioLimitArg) external;
```
Comment (tailingchen) [inline contracts/interface/IOtcMaker.sol:42]: remove admin function from interface.

File: contracts/interface/IOtcMakerStruct.sol (hunk 4 above)
Comment (DylanYang0523) [reply contracts/interface/IOtcMakerStruct.sol:1]: Nice catch!!!

**Review by tailingchen — APPROVED** [review]
